### PR TITLE
Adds initial types

### PR DESCRIPTION
**Pull request by sobolevn**
There are a lot of `TODO`s! 

Currently I am almost unable to type this code.
But, typing external API is 100% required.
For me personally this is a road-blocker to start using this library.

Any future help with the typings is appreciated. 

Global `TODO`s:
- [ ] add `flake8-pyi`
- [ ] add `mypy` 
- [ ] add tests for types, see https://github.com/dry-python/returns/issues/47

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Check http://editorconfig.org for more information
```
Comment (proofit404) [inline .editorconfig]: I agree to use editor config in all our projects. But in my opinion, it should be a separate PR. Preferably in all projects at the same time.

File: poetry.lock [hunk 2]
```diff
@@ -0,0 +1,142 @@
+[[package]]
```
Comment (proofit404) [inline poetry.lock]: `stories`, `dependencies` and `mappers` have zero dependencies. `mkdocs` extra was introduced to install the material theme on `readthedocs` service.

I know it's not documented anywhere, but I prefer not to lock developer dependencies. Our policy should be "always use the most recent version of the tool available on target python version". Dependabot and pyup are cool but it brings so much PR nose to follow the most recent version policy.

Let's remove `poetry.lock` or even include it in the `.gitignore` file.

File: src/mappers/_mapper.pyi [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    def __init__(self, *args) -> None:
+        ...
+
+    def reader(self, f: Callable) -> 'Reader':
+        ...
+
+
+# TODO: type is not clear
+# TODO: type is dependent, not sure if proper typing is possible
+def decompose(args) -> tuple:
```
Comment (proofit404) [inline src/mappers/_mapper.pyi]: Should we type public package API only?

File: .editorconfig (hunk 1 above)
Comment (sobolevn) [reply .editorconfig]: That's my IDE does by default.

File: poetry.lock (hunk 2 above)
Comment (sobolevn) [reply poetry.lock]: Ok

File: src/mappers/_mapper.pyi (hunk 3 above)
Comment (sobolevn) [reply src/mappers/_mapper.pyi]: Of course not, typing public API only is valuable to our users, while typing internal source code is valuable for developers. 

Having correct types makes programs better in three ways: less bugs, better design, and better docs.

Ideally it should be typed in-place. But, since we support `python2.7` (which is dead) it can be stored in `pyi` files.

File: .editorconfig (hunk 1 above)
Comment (proofit404) [reply .editorconfig]: Go fix your IDE first 😂

File: src/mappers/_mapper.pyi (hunk 3 above)
Comment (proofit404) [reply src/mappers/_mapper.pyi]: Python 2 does not magically disappear after 2020. I have plans to support it as long as it is possible and not harm the implementation of the core features. Also, it's possible that we will port it to Python 2.6 as part of a paid plan 😂

Is there a way to use type annotations in comments the way it will be recognized by mypy when someone uses our library in their projects?

Comment (sobolevn) [reply src/mappers/_mapper.pyi]: Yes, but for very simple cases only. Like `# type: int`
But, it does not work with `@overload`and several other features which is crucial.

**Comment by proofit404** [comment]
Will be fixed as part of the #53
